test/withdraw.test.ts

```diff
@@ -265,5 +246,118 @@ describe('StakingPool.withdraw', () => {
+        await testEnv.stakingPool
+          .connect(deployer)
+          .initNewPool(rewardPersecond, firstTimestamp, duration);
+        await testEnv.stakingAsset.connect(alice).faucet();
+        const tx = await testEnv.stakingAsset.connect(alice).approve(testEnv.stakingPool.address, RAY);
+        await advanceTimeTo(await getTimestamp(tx), firstTimestamp);
+        await testEnv.stakingPool.connect(alice).stake(amount.mul(3));
+        await testEnv.rewardAsset.connect(deployer).faucet();
+        await testEnv.rewardAsset.connect(deployer).transfer(testEnv.stakingPool.address, ethers.utils.parseEther('100'));
+      });
+
+      it('rewardPerSecond is changed and withdraw all', async () => {        
+        const poolDataBefore = await getPoolData(testEnv);
+        const userDataBefore = await getUserData(testEnv, alice);
+        const tx = await testEnv.stakingPool.connect(deployer).extendPool(rewardPersecond, duration);
+
+        const [expectedPoolData_1, expectedUserData_1] = updatePoolData(
+          poolDataBefore,
+          userDataBefore,
+          await getTimestamp(tx),
+          duration,
+          rewardPersecond
+        );
+
+        const withdrawTx = await testEnv.stakingPool
+          .connect(alice)
+          .withdraw(amount.mul(3));
+
+        const [expectedPoolData_2, expectedUserData_2] = expectDataAfterWithdraw(
+          expectedPoolData_1,
+          expectedUserData_1,
+          await getTimestamp(withdrawTx),
+          amount.mul(3)
+        );
+        const poolDataAfter = await getPoolData(testEnv);
+        const userDataAfter = await getUserData(testEnv, alice);
+
+        expect(poolDataAfter).eql(expectedPoolData_2);
+        expect(userDataAfter).eql(expectedUserData_2);
+      });
+
+      it('rewardPerSecond is changed and withdraw partial', async () => {
+        const poolDataBefore = await getPoolData(testEnv);
+        const userDataBefore = await getUserData(testEnv, alice);
+        const tx = await testEnv.stakingPool.connect(deployer).extendPool(rewardPersecond, duration);
+  
+  
+        const [expectedPoolData_1, expectedUserData_1] = updatePoolData(
+          poolDataBefore,
+          userDataBefore,
+          await getTimestamp(tx),
+          duration,
+          rewardPersecond
+        );
+  
+        const withdrawTx = await testEnv.stakingPool
+          .connect(alice)
+          .withdraw(amount);
+  
+        const [expectedPoolData_2, expectedUserData_2] = expectDataAfterWithdraw(
+          expectedPoolData_1,
+          expectedUserData_1,
+          await getTimestamp(withdrawTx),
+          amount
+        );
+        const poolDataAfter = await getPoolData(testEnv);
+        const userDataAfter = await getUserData(testEnv, alice);
+  
+        expect(poolDataAfter).eql(expectedPoolData_2);
+        expect(userDataAfter).eql(expectedUserData_2);
+      });
+
+      it('rewardPerSecond is changed and stake and withdraw', async () => {
```

Comment:
I understand why you take expectedPoolData_1 and expectedUserData_1 as a parameter in expectedDataAfterStake, but I don't think that expectedDataAfterWithdraw takes the `expected User, Pool Data` former function does. 
I think If the intent was to check the scenario in which the user stakes, withdraws, and then claims, the `expectedAfter` function's parameter should be actual contract data.